File: 1512 Num Of Good Pairs/numOfGoodPairs_nested_loop.py
# Double iteration, compare from right to left
# Ignore last index in vec/array since theres nothing to compare. 
# Time : O(n^2) since we are using nested loop...
# Beat 80%. 37ms, 16.28mb


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count = 0
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                count += 1 if nums[i] == nums[j] else 0
        return count

File: 1512 Num Of Good Pairs/numOfGoodPairs_nested_loop.java
// Double iteration, compare from right to left
// Ignore last index in vec/array since theres nothing to compare. 
// Time : O(n^2) since we are using nested loop...
// Beat 86.48%. 1ms, 39.52mb

class Solution {
    public int numIdenticalPairs(int[] nums) {
        int count = 0;
        
        for (int i = 0; i < nums.length - 1; i++)
        {
            for (int j = i + 1; j < nums.length; j++)
            {
                if (nums[i] == nums[j])
                {
                    count++;
                }
            }
        }
        return count;
    }
}